hdmi/models/encoder_model.py

```diff
@@ -0,0 +1,271 @@
+                         '1010101011']  # 11
+
+        video_guard_band = {
+            'BLUE': int('1011001100', 2),
+            'GREEN': int('0100110011', 2)
+        }.get(self.channel, int('1011001100', 2))
+
+        data_island_guard_band = {
+            'GREEN': int('0100110011', 2),
+            'RED': int('0100110011', 2)
+        }.get(self.channel, 0)
+
+        no_of_ones_video_in = Signal(intbv(0)[math.log(self.color_depth, 2):])
+
+        decision1 = Signal(bool(0))
+        decision2 = Signal(bool(0))
+        decision3 = Signal(bool(0))
+
+        # input video delayed by a clock cycle
+        _video_in = Signal(intbv(0, min=self.video_in.min,
+                                 max=self.video_in.max))
+
+        # 1 bit more than the input (Signal after first stage of encoding the input)
+        q_m = Signal(intbv(0, min=self.video_in.min,
+                           max=self.video_in.max * 2))
+
+        no_of_ones_q_m = Signal(intbv(0)[math.log(self.color_depth, 2)+1:])
+        no_of_zeros_q_m = Signal(intbv(0)[math.log(self.color_depth, 2)+1:])
+
+        count = Signal(intbv(0)[5:0])
+
+        # delayed versions of vde signal
+        _vde, __vde = [Signal(bool(0)) for _ in range(2)]
+
+        # delayed versions of ade signal
+        _ade, __ade, ___ade, ____ade = [Signal(bool(0)) for _ in range(4)]
+
+        # delayed versions of c0 signal
+        _c0, __c0 = [Signal(bool(0)) for _ in range(2)]
+
+        # delayed versions of c1 signal
+        _c1, __c1 = [Signal(bool(0)) for _ in range(2)]
+
+        # delayed versions of audio_in signal
+        _audio_in, __audio_in = [Signal(intbv(0, min=self.audio_in.min,
+                                              max=self.audio_in.max)) for _ in range(2)]
+
+        _q_m = Signal(intbv(0, min=self.video_in.min,
+                            max=self.video_in.max * 2))
+
+        # Digital island guard band period
+        digb_period = Signal(bool(0))
+
+        ade_vld = Signal(bool(0))
+        audio_in_vld = Signal(intbv(0, min=self.audio_in.min,
+                                    max=self.audio_in.max))
+
+        @always(self.clock.posedge)
+        def sequential_logic():
+
+            no_of_ones_video_in.next = bin(self.video_in).count("1")
+            _video_in.next = self.video_in
+            no_of_ones_q_m.next = bin(q_m).count("1")
+            no_of_zeros_q_m.next = 8 - bin(q_m).count("1")
+
+            _vde.next = self.vde
+            __vde.next = _vde
+
+            _ade.next = self.ade
+            __ade.next = _ade
+            ___ade.next = __ade
+            ____ade.next = ___ade
+
+            _c0.next = self.c0
+            __c0.next = _c0
+            _c1.next = self.c1
+            __c1.next = _c1
+
+            _audio_in.next = self.audio_in
+            __audio_in.next = _audio_in
+
+            _q_m.next = q_m
+
+        @always(____ade, self.ade, __ade, no_of_ones_video_in, _video_in, count, no_of_ones_q_m, no_of_zeros_q_m, q_m,
```

Comment:
@cfelton Some days ago I told you that at one place `always_comb` didn't give the expected output, it was here and so I used `always` decorator, and moreover since I m both changing and accessing the Signal value, which is not possible in a single function using `always_comb` , this seemed like a better idea.